spring-ai-core/src/main/java/org/springframework/ai/document/DefaultContentFormatter.java

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright 2023-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.ai.document;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.springframework.util.Assert;
+
+/**
+ * @author Christian Tzolov
+ */
+public class DefaultContentFormatter implements ContentFormatter {
+
+	private static final String TEMPLATE_CONTENT_PLACEHOLDER = "{content}";
+
+	private static final String TEMPLATE_METADATA_STRING_PLACEHOLDER = "{metadata_string}";
+
+	private static final String TEMPLATE_VALUE_PLACEHOLDER = "{value}";
+
+	private static final String TEMPLATE_KEY_PLACEHOLDER = "{key}";
+
+	private static final String DEFAULT_METADATA_TEMPLATE = String.format("%s: %s", TEMPLATE_KEY_PLACEHOLDER,
+			TEMPLATE_VALUE_PLACEHOLDER);
+
+	private static final String DEFAULT_METADATA_SEPARATOR = "\n";
+
+	private static final String DEFAULT_TEXT_TEMPLATE = String.format("%s\n\n%s", TEMPLATE_METADATA_STRING_PLACEHOLDER,
+			TEMPLATE_CONTENT_PLACEHOLDER);
+
+	/**
+	 * Template for how metadata is formatted, with {key} and {value} placeholders.
+	 */
+	private final String metadataTemplate;
+
+	/**
+	 * Separator between metadata fields when converting to string.
+	 */
+	private final String metadataSeparator;
+
+	/**
+	 * Template for how Document text is formatted, with {content} and {metadata_string}
+	 * placeholders.
+	 */
+	private final String textTemplate;
+
+	private final ContentFormatter.MetadataMode metadataMode;
+
+	/**
+	 * Metadata keys that are excluded from text for the LLM.
+	 */
+	private final List<String> excludedLlmMetadataKeys;
```

Comment:
similar comment wrt to naming , `excludedInferenceMetadataKeys` ?

Reply:
done